### DISCUSSION TITLE
Sensor in namespace separate from EventBus and Event Source

### DISCUSSION THREAD

**Discussion by nheitz** [argoproj/argo-events#1259] [Q&A]
....is this possible?  So far, I am set up with a simple following of the docs...
controllers and event bus in "argo-events" namespace
argo workflow pieces in "argo" namespace

If I create a sensor for the example "webhook" event in the "argo" namespace, I can see that the sensor controller is picking it up correctly across the cluster.  However, the sensor pod fails to deploy, and the error in the sensor controller logs suggest that the event bus is not reachable from the "argo" namespace. Is it possible to configure event bus access to the event bus in "argo-events" namespace?

**Comment by whynowy** [comment]
What is the reason you want to do that?

**Comment by nheitz** [comment]
The documentation suggests it is possible, q2 of the faq. https://argoproj.github.io/argo-events/FAQ/

Otherwise, I'm not sure what the suggested pattern of deployment is. Should all sensors be deployed in the same namespace as the event bus? Should there be a separate namespace per event source?

**Comment by winterrobert** [comment]
Ran into this as well; what is the best approach in a scenario where we want to trigger workflows across multiple namespaces?

1. Have all sensors in the argo-events namespace together with the eventbus?
2. Have multiple eventbuses, one per namespace?

**Comment by DavidGamba** [comment]
To me it is a matter of ownership.

I can manage a single Highly Available EventBus for multiple Sensors. I have that in the `argo-events` namespace.

Multiple teams can deploy their own `Sensor` in their own namespace.

You could argue that `EventSource` resources could be centrally managed and shared or they could be namespace specific. 

I wouldn't want to have to place everything in the `argo-events` namespace. 

Basically this: https://github.com/argoproj/argo-events/issues/3048
